src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -88,6 +89,20 @@ internal struct WriteStack
         /// </summary>
         public int FlushThreshold;
 
+        private PipeWriter? _pipeWriter;
+        public PipeWriter? PipeWriter
```

Comment:
Nit: most of the data stored in this type are simple fields with any validation happening in methods that try to access it. This is largely historical, but in the interest of consistency would it be possible to do the same here? FWIW the checks are redundant in the sync stream method.

Reply:
I did it like that so I didn't have to add the exception to multiple locations.